docs/distribution/README.md

```diff
@@ -0,0 +1,927 @@
+        "digest": "sha256:333cc44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b785c333c",
+        "size": 1906
+      },
+      "layers": []
+    }
+    ```
+
+    The `manifest.config` contains the following signature information:
+
+    ```json
+    {
+        "signed": {
+            "mediaType": "application/vnd.oci.image.manifest.v2+json",
+            "digest": "sha256:111ma2d22ae5ef400769fa51c84717264cd1520ac8d93dc071374c1be49a111m",
+            "size": 528,
+            "references": [
+                "registry.acme-rockets.com/net-monitor:v1"
+            ],
+            "exp": 1627555319,
+            "nbf": 1596019319,
+            "iat": 1596019319
+        },
+        "signature": {
+            "typ": "x509",
+            "sig": "UFqN24K2fLj...",
+            "alg": "RS256",
+            "x5c": [
+                "MIIDszCCApugAwIBAgIUL1anEU/..."
+            ]
+        }
+    }
+    ```
+
+**Pros with this approach:**
+
+* OCI Artifacts already supports manifest based artifacts, through the `manifest.config.mediaType`
+
+**Cons with this approach:**
+
+* Manifests have no means to reference other artifacts.
+* An alternative is required to link a target artifact with it's signature. Either through parsing the signature `manifest.config` object, or [a separate API for linking objects](#linking-signatures-to-artifacts).
+
+### Signature Persistence - Option 2: oci-index
+
+This option is similar to using oci-manifest. However, instead of parsing the signature object to determine the linkage between an artifact and signature, the `index.manifests` collection is utilized.
+
+<img src="../../media/signature-as-index.png" width=650>
+
+Example **manifests** for a container image (`net-monitor:v1`) and two signatures (**wabbit-networks**, **acme-rockets**). The signatures are persisted as OCI Indexes, with a new `index.config` object:
+
+1. **manifest digest for the `net-monitor:v1` image:** `sha256:111ma2d22ae5ef400769fa51c84717264cd1520ac8d93dc071374c1be49a111m"`  
+
+    ```JSON
+    {
+      "schemaVersion": 2,
+      "mediaType": "application/vnd.oci.image.manifest.v2+json",
+      "config": {
+        "mediaType": "application/vnd.oci.image.config.v1+json",
+        "digest": "sha256:111ca3788f3464fd9a06386c4d7a8e3018b525278ac4b9da872943d4cfea111c",
+        "size": 1906
+      },
+      "layers": [
+        {
+          "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
+          "digest": "sha256:9834876dcfb05cb167a5c24953eba58c4ac89b1adf57f28f2f9d09af107ee8f0",
+          "size": 32654
+        },
+        {
+          "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
+          "digest": "sha256:ec4b8955958665577945c89419d1af06b5f7636b4ac3da7f12184802ad867736",
+          "size": 73109
+        }
+      ]
+    }
+    ```
+
+2. **index digest for the wabbit-networks signature** `sha256:222ibbf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cb222i`
```

Comment:
Another approach, very similar to this but without taking a dependency on adding a new index `config` descriptor, would be to add the signature descriptor in the `manifests` array instead.  You could have an index that looked like this:

```json
{
    "schemaVersion": 2,
    "mediaType": "application/vnd.oci.image.index.v2+json",
    "manifests": [
        {
            "mediaType": "application/vnd.oci.image.manifest.v1+json",
            "digest": "sha256:111ma2d22ae5ef400769fa51c84717264cd1520ac8d93dc071374c1be49a111m",
            "size": 7023,
            "platform": {
                "architecture": "ppc64le",
                "os": "linux"
            }
        },
        {
            "mediaType": "application/vnd.cncf.notary.config.v2+jwt",
            "digest": "sha256:222cb130c152895905abe66279dd9feaa68091ba55619f5b900f2ebed38b222c",
            "size": 1906
        }
    ]
}
```

The client can still determine that it's looking for a signature by way of iterating through the `manifests` and finding one with the correct `mediaType`.  (This is a bit of an abuse of the name "manifests", but so is the way you're using "config" so it's not terribly different.)  This approach also gives you the opportunity to add multiple signatures in the same index.

Reply:
This is the curx of the design question:
The challenges are:
* The manifest collection is a collection of `application/vnd.oci.image.manifest.v1+json` or `application/vnd.oci.image.index.v1+json` objects. For the artifacts, we use the `config.mediaType` to identify the type. Here' we're mixing descriptor mediaTypes vs. schema mediaTypes. I kinda like your suggestion as all descriptors are currently limited to the two and this makes it a challenge to return a reverse lookup result and not know what config.mediaType (aka artifact type) is represented.  But, it's a bigger change to registry operators. However, it also means signature is just another artifact type, and we use a generic model that could support other artifact types.
* While it could support multiple signatures, that assumes the multiple signatures are uploaded at once. We've said signatures are additive, so this means adding another signature would either be an update to an index, or another index. Seems intermixing these would add complexity. It's also unclear if I can add multiple images and multiple signatures, what does that mean? Are there multiple signatures to all other artifact types in the manifest collection? 
* The main reason to add config to Index is to enable different types of Index based artifact types. For instance CNAB is a multi-artifact collection. Today, the CNAB is an index that just kinda floats out there. They use annotations, which is really hacky, and they store the artifact content in one of the images, another limitation that gets unblocked if we can say an index is a CNAB. `"index.config.mediaType=application/vnd.org.cnab.bundle.v1.config`
* The config is storing data thing is a common question. A config object is the same as a layer object. But, it's called "config" so it's weird. Should an index support a layers collection as well? How impactful would that be to registry operators?
* The purest way is the [Signature Persistence - Option 3: oci-manifest linked through oci-index](https://github.com/notaryproject/nv2/blob/276abe450feec8f3b54f0774b7dfb47f36670cc5/docs/distribution/README.md#signature-persistence---option-3-oci-manifest-linked-through-oci-index)

Since `index.config.mediaType` is consistent with `manifest.config.mediaType` to identify the artifact type, it just seems less "dirty" to store the signature as a config "layer". 
It's a toss up for how much purity, refactoring, or layering we want to implement.
Since all of this is details "within the sausage" and only a few would have to implement this, vs. how it's exposed to end users. I was leaning towards [Signature Persistence - Option 2: oci-index](https://github.com/notaryproject/nv2/blob/276abe450feec8f3b54f0774b7dfb47f36670cc5/docs/distribution/README.md#signature-persistence---option-2-oci-index). 

I do imagine as we continue sketching this out, we'll see a few more opportunities. 
For instance, you'll see we're actually linking from the artifact to the config of the signature index. Instead of the artifact linking to the index digest, which you then have to query again to get the config object.